docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,85 @@
+# ADR 56: Proving Amnesia Attack
+
+## Changelog
+
+- 02.04.20: Initial Draft
+
+## Context
+
+Whilst most created evidence of malicious behaviour is self evident such that any individual can verify them independently there are types of evidence that require collaboration from the network of validators, known as global evidence, in order to accumulate enough information to create evidence that is individually verifiable and can therefore be processed through consensus. Fork Accountability as a whole constitutes of detection, proving and punishing. This ADR addresses how to prove infringement from global evidence.
+
+So far, only [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks such as amnesia (which will be what the reactor initially focuses on) and back to the past (only causes light forks) have been identified as forms of global evidence that require more information from other validators.
+
+## Decision
+
+A new reactor will perform the task of turning global evidence to individual evidence. This special circumstance reactor could be combined with the current evidence reactor although the evidence reactor is currently used for gossiping and storing of evidence and so it's purpose would expand. The reactor is initialized when a full node receives PotentialAmnesiaEvidence from a light client.
+
+*REMARK: This means that only the set of full nodes that are directly connected to the light client that observed the misbehaviour will process the evidence. Is this sufficient, do we want to gossip to further nodes, do we want just a single node to process evidence to avoid too much communication overhead?*
+
+```
+type PotentialAmnesiaEvidence struct {
+	C1 types.Commit
+	C2 types.Commit
+}
+```
+
+The receiving full node first runs `Verify()` to ensure that the evidence is valid i.e. both `Commit`s are valid and are of the same height.
+
+*NOTE: This means that the commits should be within the bonding period minus some margin as well as within the expected range of heights wherein votesets are kept*
+
+The suspect group are the overlapping set of validators that have voted for both `C1` and `C2`.
+
+The full node requests `VoteSet`'s from all validators in that validator set. A configurable time or height is set with which validators must respond. If a validator, that is part of the suspect group, fails to provide a vote, we mark them as faulty.
```

Comment:
I think this is a good place to ask the question, are we okay with this? Is it okay to mark a non-responsive node as faulty?